concepts/&str/about.md

```diff
@@ -1,3 +1,69 @@
 # About
 
-TODO
+There are two ways that Rust implements an array of characters: [str][str] and [String][String].
+
+`str`, also known as a "string slice", is usually used with a reference as `&str`.
+
+To know why, it may help to understand why the following line of code will not compile
+
+```rust
+const my_const: str = "Hello, World!";
```

Comment:
```suggestion
const my_const: str = *"Hello, World!";
```

We'd need the dereference there to avoid a type error; string literals have type `&'static str`. For example, this is legal:

```rust
const HELLO: [u8; 12] = *b"Hello, world";
```

Reply:
The suggested line

```rust
const my_const: str = *"Hello, World!";
```
results in the same error. We need a reference, not a dereference.

Reply:
Yes, I know what you mean that the dereference is meant to bring the  `&'static str` to a str, but I'm not sure it might be confusing to throw the deref operator in there for students who need help with `&str`. I wanted to avoid getting into derefs in order to keep the focus on `&str`, but I don't feel strongly about putting the `*` in there if you think it won't confuse the matter for newer students.

Reply:
I was tempted to cover `&'static str` but decided to keep lifetimes out of it.

Reply:
I get the type mismatch error on a let, but not a const. Perhaps it should have the deref operator in case the compiler ever changes to squawk  about the mismatch. Sorry for all the thinking out loud...